Formula/ocl-icd.rb

```diff
@@ -0,0 +1,50 @@
+class OclIcd < Formula
+  desc "OpenCL ICD loader"
+  homepage "https://github.com/OCL-dev/ocl-icd/"
+  url "https://github.com/OCL-dev/ocl-icd/archive/refs/tags/v2.3.1.tar.gz"
+  sha256 "a32b67c2d52ffbaf490be9fc18b46428ab807ab11eff7664d7ff75e06cfafd6d"
+  license "BSD-2-Clause"
+  head "https://github.com/OCL-dev/ocl-icd.git", branch: "master"
+
+  keg_only :shadowed_by_macos, "macOS provides OpenCL.framework"
+
+  depends_on "asciidoc" => :build
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+  depends_on "opencl-headers" => [:build, :test]
+  depends_on "xmlto" => :build
+
+  uses_from_macos "libxml2" => :build
+  uses_from_macos "libxslt" => :build
+  uses_from_macos "ruby" => :build
+
+  def install
+    ENV["XML_CATALOG_FILES"] = etc/"xml/catalog"
+    system "./bootstrap"
+    system "./configure", *std_configure_args,
+                          "--disable-silent-rules",
+                          "--enable-custom-vendordir=#{etc}/OpenCL/vendors"
+    system "make", "install"
+    pkgshare.install "ocl_test.c"
+  end
+
+  def caveats
+    s = "The default vendors directory is #{etc}/OpenCL/vendors\n"
+    on_linux do
+      s += <<~EOS
+        No OpenCL implementation is pre-installed, so all dependents will require either
+        installing a compatible formula or creating an ".icd" file mapping to a manually
+        installed option. A portable implementation is available via the `pocl` formula.
+      EOS
+    end
```

Comment:
Officially, the specification explicitly says to use `/etc/OpenCL/vendors`, which is why I added first caveat.

The latter part is due to the fact that the intended distribution of OpenCL stack differs from other formulae where I haven't pre-installed an OpenCL implementation/runtime. On macOS, the OpenCL framework plays this role.

Alternatively, we could force install `pocl` to get runtime functionality; however, this may not be useful depending on user's hardware as Intel/AMD/Nvidia implementations are better tuned to specific hardware. At least some Intel iGPU & AMD GPU have open-source options. Nvidia is mainly proprietary.

Reply:
As I understand it, if we install `pocl` this doesn't preclude the user installing other OpenCL backends and using those instead.  They just have to configure everything so that their backend is used instead of `pocl`.  If that is correct, then I think we can just always install `pocl` and update the caveat to say that we have a shipped a very basic OpenCL backend for testing purposes but the user should install a higher performance backend tuned to their specific hardware for more intensive applications.

The Intel and AMD implementations are open source and do appear compatible with Homebrew (possibly aside from notability requirements), but they could be difficult to test in a CI environment, so it may not be feasible for us to ship them.  

As long as our caveat makes it clear that the user will need to take further action to get a performant OpenCL implementation, I think our approach to this will be acceptable.